core/include/vecmem/memory/resources/memory_resource.hpp

```diff
@@ -0,0 +1,35 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+/*
+ * The purpose of this file is to provide uniform access (on a source-code
+ * level) to the memory_resource and polymorphic_allocator types from the
+ * standard library. These are either in the std::pmr namespace or in the
+ * std::experimental::pmr namespace depending on the GCC version used, so we try
+ * to unify them by aliassing depending on the compiler feature flags.
+ */
+#if __has_include(<memory_resource>)
+#include <memory_resource>
+
+namespace vecmem::resource {
+    using memory_resource = std::pmr::memory_resource;
+
+    template<typename T>
+    using polymorphic_allocator = std::pmr::polymorphic_allocator<T>;
+}
```

Comment:
I would prefer the simpler `vecmem::memory_resource` and `vecmem::polymorphic_allocator` names. Since eventually these will be called `std::memory_resource` and `std::polymorphic_allocator` (at least I'm pretty sure that they will be...), we shouldn't try to make the name more elaborate than it needs to be...